tests/test_runs/test_run_functions.py

```diff
@@ -1336,3 +1333,44 @@ def test_run_flow_on_task_downloaded_flow(self):
         run.publish()
         TestBase._mark_entity_for_removal("run", run.run_id)
         TestBase.logger.info("collected from {}: {}".format(__file__.split("/")[-1], run.run_id))
+
+    def test_format_prediction_non_supervised(self):
+        # non-supervised tasks don't exist on the test server
+        openml.config.server = self.production_server
+        clustering = openml.tasks.get_task(126033, download_data=False)
+        ignored_input = [0] * 5
+        with self.assertRaises(TypeError):
+            format_prediction(clustering, *ignored_input)
+
+    def test_format_prediction_classification_no_probabilities(self):
+        classification = openml.tasks.get_task(self.TEST_SERVER_TASK_SIMPLE[0], download_data=False)
+        ignored_input = [0] * 5
+        with self.assertRaises(ValueError):
```

Comment:
I'd actually prefer to have `self.assertRaisesRegex` so that one sees what exact error is expected. Could you please change that?

Reply:
I've actually been meaning to ask. Why do we use the `unittest` module for some things, and `pytest` for others?
For instance, instead of `self.assertRaisesRegex`  you would use `pytest.raises(ValueError, match=r'regex string')`.

Reply:
Historical reasons ;)

I think one could actually only use `assert a == b` and not use the `unittest` or `pytest` module any more, see https://docs.pytest.org/en/stable/assert.html

Reply:
Yes, but just `assert` does not work for checking exceptions. The page you linked describes the `pytest.raises(ValueError, match=r'regex string')` call I referred too as well.

Reply:
I'll use the `unittest` `self.assertRaisesRegex` for consistency.